### PR TITLE
Initial tests for the Heading component

### PR DESCRIPTION
**Pull request by wove5**
Awaiting review by Samitha and Walter.

### CODE REVIEW

**Comment by wove5** [comment]
Went back and applied  "format:fix"  and  "lint:fix" in new commit, then closed and reopened the PR after which auto checks passed.

File: components/heading/tests/unit/heading.spec.js [hunk 1]
```diff
@@ -1,3 +1,43 @@
...
-	expect(true).toBe(true);
+import React from 'react';
+import { Heading } from '@westpac/heading';
+import { render } from '@testing-library/react';
+
+import { overridesTest } from '../../../../helpers/tests/overrides-test.js';
+import { nestingTest } from '../../../../helpers/tests/nesting-test.js';
+import { GEL } from '@westpac/core';
+import wbc from '@westpac/wbc';
+// import { Heading } from '@westpac/button-dropdown';
```
Comment (wpalombini) [inline components/heading/tests/unit/heading.spec.js:9]: If this is not going to be used, please remove it, thanks.

File: components/heading/tests/unit/heading.spec.js [hunk 2]
```diff
@@ -1,3 +1,43 @@
...
+
+// The default tests every component should run
+overridesTest({
+	name: 'heading', // the name has to be the package name without '@westpac/' scope
+	overrides: ['Heading'], // every single override root key
```
Comment (wpalombini) [inline components/heading/tests/unit/heading.spec.js:14]: It seems that we are missing an override here: BrandHeading. Check the components/heading/src/overrides folder. You can also compare this with the Alert tests.

**Review by wpalombini — COMMENTED** [review]
Hi David, thanks for working on adding more tests to it. I just let a couple of comments, otherwise, it looks good to me.

File: components/heading/tests/unit/heading.spec.js (hunk 1 above)
Comment (wove5) [reply components/heading/tests/unit/heading.spec.js:9]: Yes, shall do - look for next commit.  Line 9 came up in auto completion during coding, then I commented it out - forgot to remove it.

File: components/heading/tests/unit/heading.spec.js (hunk 2 above)
Comment (wove5) [reply components/heading/tests/unit/heading.spec.js:14]: overrides: ['BrandHeading', 'Heading'], // every single override root key
This is making the unit test fail. I need to discuss this with you further.

Comment (wove5) [reply components/heading/tests/unit/heading.spec.js:13]: Ok, now the branch includes testing for BrandHeading component.

**Review by wpalombini — APPROVED** [review]
It looks good David!

**Pull request merged**